libraries/AP_HAL_PX4/I2CDevice.cpp

```diff
@@ -79,8 +79,8 @@ bool PX4_I2C::do_transfer(uint8_t address, const uint8_t *send, uint32_t send_le
         init_done = true;
         // we do late init() so we can setup the device paths
         
-        snprintf(devname, sizeof(devname), "AP_I2C_%u", instance);
-        snprintf(devpath, sizeof(devpath), "/dev/api2c%u", instance);
+        snprintf(devname, sizeof(devname), "AP_I2C_%1u", instance);
```

Comment:
What does %1u do different than just %u?

Reply:
sorry, leftover of playing around. Fixed in a second...